Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py

```diff
@@ -474,28 +536,61 @@ def get_indicators(indicators: str) -> list:
+    for i, single_batch_iocs in enumerate(batch_iocs(iocs, batch_size=MAX_INDICATORS_TO_SYNC)):
+        demisto.debug(f'pushing IOCs to XDR: batch #{i} with {len(single_batch_iocs)} IOCs')
+        requests_kwargs: dict = get_requests_kwargs(_json=list(
+            map(demisto_ioc_to_xdr, single_batch_iocs)), validate=True)
+        response = client.http_request(url_suffix=path, requests_kwargs=requests_kwargs)
+        validation_errors.extend(response.get('reply', {}).get('validation_errors'))
+    return validation_errors
+
+
+def push_indicators_to_xdr_request(client, indicators, ):
+    path = 'tim_insert_jsons/'
+    demisto.debug(f"pushing IOCs to XDR: pushing {len(indicators)} IOCs to the {path} endpoint")
+    requests_kwargs: dict = get_requests_kwargs(_json=indicators, validate=True)
+    response = client.http_request(url_suffix=path, requests_kwargs=requests_kwargs)
+    if response.get('reply', {}).get('success') is not True:
+        raise DemistoException(f"Response status was not success, {response=}")
+    return response
+
+
 def tim_insert_jsons(client: Client):
-    # takes our changes and pushes to XDR
+    # Retrieve iocs changes from xsoar and pushes to XDR
     indicators = demisto.args().get('indicator', '')
+    validation_errors: list = []
+    # If tim_insert_jsons is called from xdr-iocs-push is called
     if indicators:
         demisto.info(f"pushing IOCs to XDR: querying with input {indicators}")
         iocs = get_indicators(indicators)
+        if iocs:
+            validation_errors = push_iocs(client, iocs)
+        else:
+            demisto.info("pushing IOCs to XDR: found no matching IOCs")
+    # If tim_insert_jsons is called from fetch_indicators
     else:
         demisto.info("pushing IOCs to XDR: did not get indicators, will use recently-modified IOCs")
-        iocs = get_last_iocs()
-
-    validation_errors = []
-    if iocs:
-        path = 'tim_insert_jsons/'
-        demisto.info(f"pushing IOCs to XDR: pushing {len(iocs)} IOCs to the {path} endpoint")
-        for i, single_batch_iocs in enumerate(batch_iocs(iocs)):
-            demisto.debug(f'pushing IOCs to XDR: batch #{i} with {len(single_batch_iocs)} IOCs')
-            requests_kwargs: dict = get_requests_kwargs(_json=list(
-                map(demisto_ioc_to_xdr, single_batch_iocs)), validate=True)
-            response = client.http_request(url_suffix=path, requests_kwargs=requests_kwargs)
-            validation_errors.extend(response.get('reply', {}).get('validation_errors'))
-    else:
-        demisto.info("pushing IOCs to XDR: found no matching IOCs")
+        current_run: str = datetime.utcnow().strftime(DEMISTO_TIME_FORMAT)
+        integration_context: dict = get_integration_context()
+        while True:
+            query = (create_query_with_end_time(to_date=current_run)
+                     if integration_context.get('search_after')
```

Comment:
lets talk about it

Reply:
changed to:
        current_run: str = datetime.utcnow().strftime(DEMISTO_TIME_FORMAT)
        while True:
            last_run: dict = get_integration_context()
            query = (create_query_with_end_time(to_date=current_run)